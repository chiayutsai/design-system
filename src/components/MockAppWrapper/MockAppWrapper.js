import history from 'history'
import PropTypes from 'prop-types'
import App from 'components/App'
import configureStore from 'store/configureStore'
import { injectReducer } from 'store/makeRootReducer'

const initialState = {}

export const store = configureStore(initialState, { history })

const MockAppWrapper = ({ children }) => (
  <App
    context={{
      pathname: '',
      store,
      injectReducer,
    }}
    insertCss={() => {}}>
    {children}
  </App>
)

MockAppWrapper.propTypes = {
  children: PropTypes.node.isRequired,
}

export default MockAppWrapper
