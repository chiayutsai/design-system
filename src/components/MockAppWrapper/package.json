{
  "name": "MockAppWrapper",
  "version": "0.0.0",
  "private": true,
  "main": "./MockAppWrapper.js"
}
