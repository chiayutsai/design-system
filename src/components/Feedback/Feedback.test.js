import renderer from 'react-test-renderer'
import MockAppWrapper from 'components/MockAppWrapper'
import Feedback from './Feedback'

describe('Feedback', () => {
  test('render correctly', () => {
    const wrapper = renderer
      .create(
        <MockAppWrapper>
          <Feedback />
        </MockAppWrapper>,
      )
      .toJSON()

    expect(wrapper).toMatchSnapshot()
  })
})
