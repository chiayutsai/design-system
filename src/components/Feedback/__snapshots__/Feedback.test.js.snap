// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback render correctly 1`] = `
<div
  className="root"
>
  <div
    className="container"
  >
    <a
      className="link"
      href="https://gitter.im/kriasoft/react-starter-kit"
    >
      Ask a question
    </a>
    <span
      className="spacer"
    >
      |
    </span>
    <a
      className="link"
      href="https://github.com/kriasoft/react-starter-kit/issues/new"
    >
      Report an issue
    </a>
  </div>
</div>
`;
